### PR TITLE
Preloading critical style-sheets for improved performance 🚤  

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
This is inspired from `Next.js`.
[Preload critical stylesheets to improve loading speed](https://web.dev/preload-critical-assets/).

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

_Sorry if I made any mistakes :(_

### CODE REVIEW

File: packages/kit/src/runtime/server/page/render.js [hunk 1]
```diff
@@ -248,6 +248,7 @@ export async function render_response({
 		}
 
 		attributes.unshift('rel="stylesheet"');
+		head += `\n\t<link rel="preload" href="${path}" as="style">`;
```
Comment (rcabralc) [inline packages/kit/src/runtime/server/page/render.js:251]: I think this preloading is already being done by the Link header. Maybe the header should be replaced with this (might benefit prerendered pages)? But only if it's not inlined, I guess.

Comment (sanjaiyan-dev) [reply packages/kit/src/runtime/server/page/render.js:251]: > I think this preloading is already being done by the Link header. Maybe the header should be replaced with this (might benefit prerendered pages)? But only if it's not inlined, I guess.

Does adding both have any drawbacks ?

Comment (benmccann) [reply packages/kit/src/runtime/server/page/render.js:251]: > Maybe the header should be replaced with this (might benefit prerendered pages)?

The header is better performance. We fallback to including in the page when prerendering: https://github.com/sveltejs/kit/blob/56cb62dcbad93b6317f98aa93c87f469f0b199eb/packages/kit/src/runtime/server/page/render.js#L260

**Comment by benmccann** [comment]
This isn't really mergable and there isn't enough detail in the issue description to understand if you're actually improving anything. It seems like you're mostly just duplicating existing functionality. If there's some issue with the existing implementation I'd recommend first filing an issue so we can get a clear understanding of what the issue is

**Pull request closed without merging**